sound/soc/sof/loader.c

```diff
@@ -280,7 +280,8 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
 
 	/* create fw_version debugfs to store boot version info */
 	if (sdev->first_boot) {
-		ret = snd_sof_debugfs_create_item(sdev, &sdev->fw_version,
+		ret = snd_sof_debugfs_create_item(sdev, (void __iomem *)
```

Comment:
@plbossart i'm not sure this is correct. I am not sure if we can annotate sdev->fw_version with a __iomem since it isnt really IO memory.

I think the underlying problem might be the signature for the snd_sof-debugfs_create_item() method. I think it was originally intended to  create debugfs entries for the memory windows which are IO memory. I think we should change it to take just a void *.

Reply:
@ranj063 I believe fw_version is part of the IPC reply, so will be in SRAM windows.
@xiulipan can you please comment, I think you're the one who added this code?

Reply:
@plbossart @ranj063 
I think @ranj063 is right about the `sdev->fw_version `, this structure want to save a copy of fw_version in kernel memory. So this buffer is not some  __iomem.
So I think the warning from spares about `snd_sof_debugfs_create_item` is that it is designed to map to some IO memory, but the `sdev->fw_version` is not.
I think I may misuse the `snd_sof_debugfs_create_item`, I should just call some debufs function to create this simple item.